src/app/libp2p_helper/src/codanet.go

```diff
@@ -465,25 +465,25 @@ func (cv customValidator) Select(key string, values [][]byte) (int, error) {
 }
 
 func (h *Helper) handlePxStreams(s network.Stream) {
-	fromSeed := false
-
-	for _, seed := range h.Seeds {
-		if s.Conn().RemotePeer() == seed.ID {
-			fromSeed = true
-			break
-		}
-	}
-
-	if !fromSeed {
-		logger.Debugf("ignoring peer-exchange stream from non-seed peer=%s", s.Conn().RemotePeer())
+	connInfo := h.ConnectionManager.GetInfo()
+	if connInfo.ConnCount >= connInfo.LowWater {
 		_ = s.Close()
 		return
 	}
 
 	for {
-		dec := json.NewDecoder(s)
+		buf := make([]byte, 8192)
```

Comment:
Does this reallocate `buf` on every iteration of the loop? Might be better to move this outside of the loop.

Reply:
Come to think of this... why is this even a loop?